packages/@react-spectrum/combobox/stories/ComboBox.stories.tsx

```diff
@@ -939,14 +939,29 @@ let ControlledValueOpenCombobox = (props) => {
   };
 
   let onSelectionChange = (key) => {
-    setFieldState({
-      isOpen: false,
-      inputValue: items.find(item => item.id === key)?.name ?? ''
-    });
+    let newInputValue = items.find(item => item.id === key)?.name;
+    setFieldState(prevState => ({
+      // If key is null (aka selected key is being cleared), preserve previous open state
+      // e.g. if user clears input field, previous open state will be open then this onSelectionChange will fire for the selected key removal. We'll want to keep that open state
+      isOpen: key == null ? prevState.isOpen : false,
+      inputValue: newInputValue ?? (props.allowsCustomValue ? prevState.inputValue : '')
+    }));
+  };
+
+  // A bit annoying but can't figure out an alternative here. Without tracking focus state manually,
+  // onSelectionChange doesn't have enough information to set the proper open state.
+  // These two cases that are seen as the same from inside onSelectionChange but want to have two different final open states:
+  // 1. User has a selected key and then deletes all input text. This causes onInputChange to fire -> isOpen = true and inputValue = "" -> onSelectionChange fires to remove selectedKey -> prevInputValue = "", prevIsOpen = true, key = null -> desired end state is to keep isOpen as true
+  // 2. User doesn't have a selected key but has the menu open and blurs. This causes onSelectionChange to fire with null  -> prevInputValue = "", prevIsOpen = true, key = null -> desired end state is to keep isOpen as false
+  let onFocusChange = (isFocused) => {
+    setFieldState(prevState => ({
+      isOpen: isFocused ? prevState.isOpen : false,
+      inputValue: prevState.inputValue
+    }));
```

Comment:
Bit iffy about requiring the user to track focus just to properly set the menu open state when deleting the input in a controlled open state + key combobox but this was the best solution I could come up with. Open to suggestions